### ISSUE TITLE
Bug ? SC2086 - false positiv due to bash arithmetic ?!

### ISSUE THREAD

**Issue by PeterPitterling** [koalaman/shellcheck#2364]
#### For bugs
- Rule Id SC2086
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

Maybe similar to ["retryLimit appears unused" but its used.](https://github.com/koalaman/shellcheck/issues/2186)

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash

declare -i prefix_length=0

(( prefix_length += 17 ))                      #triggers SC2086 in next line

printf '%-*b\n' ${prefix_length} 'same string with long prefix'
```

#### Here's what shellcheck currently says:
```console
Line 7:
printf '%-*b\n' ${prefix_length} 'same string with long prefix'
                ^-- SC2086 (info): Double quote to prevent globbing and word splitting.

Did you mean: (apply this, apply all SC2086)
printf '%-*b\n' "${prefix_length}" 'same string with long prefix'
```

#### Here's what I wanted or expected to see:
No error message

**Comment by koalaman** [comment]
ShellCheck really needs to start understanding `declare -i`, but that's a bigger job. For now I just updated this naive logic to realize that arithmetic assignments will cause the value not to contain spaces. Thanks!

**Comment by PeterPitterling** [comment]
how long does it take to materialize on shellcheck.net ?

**Comment by koalaman** [comment]
Huh, the tests passed locally but not on the host so it didn't update. I'll have a look later today.

**Comment by PeterPitterling** [comment]
same issue with 

**((prefix_length++))**

while this is accepted now ...
((prefix_length+=1))

**Comment by ramses0** [comment]
(please disregard reference from https://github.com/koalaman/shellcheck/issues/2687, it was a typo of 2346 v. 2364)
